pkgs/development/python-modules/lastversion/default.nix

```diff
@@ -0,0 +1,113 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+  # nativeBuildInputs
+, pythonRelaxDepsHook
+, setuptools
+  # nativeCheckInputs
+, pytestCheckHook
+  # install_requires
+, appdirs
+, beautifulsoup4
+, cachecontrol
+, distro
+, feedparser
+, lockfile
+, packaging
+, python-dateutil
+, pyyaml
+, requests
+, tqdm
+, urllib3
+  # test_requires
+, pytest
+, flake8
+, flake8-bugbear
+, pytest-xdist
+, pytest-cov
+  # doc_requires
+, mkdocs
+, mkdocs-material
+, mkdocstrings
+, mkdocstrings-python
+}:
+buildPythonPackage rec {
+  pname = "lastversion";
+  version = "3.5.0";
+  pyproject = true;
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "dvershinin";
+    repo = "lastversion";
+    rev = "v${version}";
+    hash = "sha256-SeDLpMP8cF6CC3qJ6V8dLErl6ihpnl4lHeBkp7jtQgI=";
+  };
+
+  nativeBuildInputs = [
+    pythonRelaxDepsHook
+    setuptools
+  ];
+
+  propagatedBuildInputs = [
+    appdirs
+    beautifulsoup4
+    cachecontrol
+    cachecontrol.optional-dependencies.filecache
+    distro
+    feedparser
+    packaging
+    python-dateutil
+    pyyaml
+    requests
+    tqdm
+    urllib3
+  ];
+
+  pythonImportsCheck = [ "lastversion" ];
+
+  pythonRelaxDeps = [
+    # install_requires
+    "cachecontrol" # Use newer cachecontrol that uses filelock instead of lockfile
+    "urllib3" # The cachecontrol and requests incompatibility issue is closed
+    # tests_requires
+    "pytest-xdist" # The upstream issue is closed
+    # docs_requires
+    "mkdocs"
+    "mkdocs-material"
+  ];
+
```

Comment:
Why are we not executing tests?

Reply:
The functionality tests require internet access, as `lastversion` gets the latest version by querying the sites.

Reply:
Update: Just found some tests runnable without internet connection. I'll update to include them.